aptos-move/move-examples/token_objects/knight/sources/knight.move

```diff
@@ -0,0 +1,541 @@
+    use std::error;
+    use std::option;
+    use std::string::{Self, String};
+    use std::signer;
+    use aptos_std::string_utils;
+    use aptos_framework::object::{Self, Object};
+    use aptos_framework::event;
+    use aptos_framework::fungible_asset::{Self, Metadata};
+    use aptos_framework::primary_fungible_store;
+    use aptos_token_objects::token;
+    use aptos_token_objects::collection;
+    use aptos_token_objects::property_map;
+
+    /// The token does not exist
+    const ETOKEN_DOES_NOT_EXIST: u64 = 1;
+    /// The provided signer is not the creator
+    const ENOT_CREATOR: u64 = 2;
+    /// Attempted to mutate an immutable field
+    const EFIELD_NOT_MUTABLE: u64 = 3;
+    /// Attempted to burn a non-burnable token
+    const ETOKEN_NOT_BURNABLE: u64 = 4;
+    /// Attempted to mutate a property map that is not mutable
+    const EPROPERTIES_NOT_MUTABLE: u64 = 5;
+    // The collection does not exist
+    const ECOLLECTION_DOES_NOT_EXIST: u64 = 6;
+
+    /// The food collection name
+    const FOOD_COLLECTION_NAME: vector<u8> = b"Food Collection Name";
+    /// The food collection description
+    const FOOD_COLLECTION_DESCRIPTION: vector<u8> = b"Food Collection Description";
+    /// The food collection URI
+    const FOOD_COLLECTION_URI: vector<u8> = b"Food Collection URI";
+
+    /// The knight token collection name
+    const KNIGHT_COLLECTION_NAME: vector<u8> = b"Knight Collection Name";
+    /// The knight collection description
+    const KNIGHT_COLLECTION_DESCRIPTION: vector<u8> = b"Knight Collection Description";
+    /// The knight collection URI
+    const KNIGHT_COLLECTION_URI: vector<u8> = b"Knight Collection URI";
+
+    /// The corn token name
+    const CORN_TOKEN_NAME: vector<u8> = b"Corn Token";
+    /// The meat token name
+    const MEAT_TOKEN_NAME: vector<u8> = b"Meat Token";
+
+    /// The condition of a knight
+    const CONDITION_HUNGRY: vector<u8> = b"Hungry";
+    const CONDITION_GOOD: vector<u8> = b"Good";
+
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Knight token
+    struct KnightToken has key {
+        /// Used to mutate the token uri
+        mutator_ref: token::MutatorRef,
+        /// Used to burn.
+        burn_ref: token::BurnRef,
+        /// Used to mutate properties
+        property_mutator_ref: property_map::MutatorRef,
+        /// Used to emit LevelUpdateEvent
+        health_update_events: event::EventHandle<HealthUpdateEvent>,
+        /// the base URI of the token
+        base_uri: String,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// The knight's health point
+    struct HealthPoint has key {
+        value: u64,
+    }
+
+    /// The health update event
+    struct HealthUpdateEvent has drop, store {
+        old_health: u64,
+        new_health: u64,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    // Food Token
+    struct FoodToken has key {
```

Comment:
I think it'd be cleaner to separate food token into a separate file. We'll have two files - knight.move and food.move

Reply:
Done!